pkg/reconciler/serverlessservice/resources/services.go

```diff
@@ -33,16 +33,39 @@ const (
 	servicePortNameH2C   = "http2"
 
 	// ServicePort is the external port of the service
-	servicePort = int32(80)
+	servicePort              = int32(80)
+	serviceHTTP2Port         = int32(81)
+	activatorTargetHTTP1Port = 8080
+	activatorTargetHTTP2Port = 8081
 
 	// RequestQueuePortName specifies the port name to use for http requests
 	// in queue-proxy container.
 	requestQueuePortName string = "queue-port"
 )
 
+func pubTargetPort(sks *v1alpha1.ServerlessService, hasBackends bool) intstr.IntOrString {
+	switch {
+	case sks.Spec.Mode == v1alpha1.SKSOperationModeServe && hasBackends:
+		return intstr.FromString(requestQueuePortName)
+	case sks.Spec.Mode == v1alpha1.SKSOperationModeProxy || !hasBackends:
+		if sks.Spec.ProtocolType == networking.ProtocolH2C {
+			return intstr.FromInt(activatorTargetHTTP2Port)
+		}
+		return intstr.FromInt(activatorTargetHTTP1Port)
+	}
+	panic("webhook validation failed")
+}
+
+func pubServicePort(sks *v1alpha1.ServerlessService) int32 {
+	if sks.Spec.ProtocolType == networking.ProtocolH2C {
+		return serviceHTTP2Port
+	}
+	return servicePort
```

Comment:
This bakes in some knowledge of Istio (if I'm not mistaken).  @nghia any pointers on how we might avoid this?

Reply:
... or maybe not.  I was thinking this was the string `http2`...

Reply:
We have both. This is mostly match Activator port. Otherwise we cannot implement flawless handover between the active and passive.

Reply:
#3874 for this one, though it might remain as is, if istio still doesn't support one port with 2 protocols